src/xunit.analyzers/Descriptors.cs

```diff
@@ -254,7 +254,10 @@ internal static class Descriptors
             "Keep precision in range {0} when asserting equality of {1} typed actual value.",
             Categories.Assertions, DiagnosticSeverity.Error, isEnabledByDefault: true);
 
-        // Placeholder for rule X2017
+        internal static DiagnosticDescriptor X2017_AssertIsTypeShouldNotBeUsedForAbstractType { get; } = new DiagnosticDescriptor("xUnit2017",
```

Comment:
@marcind 

> Without a centralized place it would be too easy to have multiple rules use the same id, potentially confusing fixers, ignore pragmas, etc.

Other projects (e.g. [StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers)) prefix the name of each analyzer with the ID, ensuring that doesn't happen. That also lends itself to being easier to find the file of the analyzer associated with a warning-- people can just go to GitHub, start the fuzzy finder, and type in the ID rather than guessing at the name of the file.